### PR TITLE
Make sure fontMasterID lib keys in UFOs are unique

### PR DESCRIPTION
**Pull request by madig**
Fixes https://github.com/googlei18n/glyphsLib/issues/400.

### CODE REVIEW

File: Lib/glyphsLib/builder/masters.py [hunk 1]
```diff
@@ -90,6 +90,13 @@ def to_glyphs_master_attributes(self, source, master):
         # GSFontMaster has a random id by default
         pass
 
+    if master.id.lower() in (m.id.lower() for m in self.font.masters):
+        raise ValueError(
+            "{} contains a '{}' lib key with the duplicate value '{}'. All given masters must have a unique ID or data will get corrupted. Please check for this lib key and either remove it (will be regenerated) or change the value. If there is no key, you just witnessed something unlikely.".format(
```
Comment (anthrotype) [inline Lib/glyphsLib/builder/masters.py]: you should split these long lines manually, black doesn't help there

Comment (madig) [reply Lib/glyphsLib/builder/masters.py]: This is something that black should have one day, I need it constantly 😁

File: tests/builder/builder_test.py [hunk 2]
```diff
@@ -949,6 +949,28 @@ def test_italic_angle(self):
...
+        master2.ascender = 0
+        master2.capHeight = 0
+        master2.descender = 0
+        master2.xHeight = 0
+        font.masters.append(master2)
+        ufos = to_ufos(font, minimize_glyphs_diffs=True)
+
+        try:
+            to_glyphs(ufos)
+        except Exception as e:
```
Comment (anthrotype) [inline tests/builder/builder_test.py]: why do that? you want to hide the traceback? I'd say let it raise

Comment (madig) [reply tests/builder/builder_test.py]: This is a should-not-raise test :) I copy-pasted the concept from https://stackoverflow.com/questions/4319825/python-unittest-opposite-of-assertraises.

Comment (anthrotype) [reply tests/builder/builder_test.py]: Ok

**Pull request merged**